ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -292,29 +276,61 @@ private void handleNewBlockHashesFromNetwork(final EthMessage message) {
       return CompletableFuture.completedFuture(block);
     }
 
-    ethContext.getScheduler().scheduleSyncWorkerTask(() -> validateAndBroadcastBlock(block));
+    final BlockHeader parent =
+        protocolContext
+            .getBlockchain()
+            .getBlockHeader(block.getHeader().getParentHash())
+            .orElseThrow(
+                () ->
+                    new IllegalArgumentException(
+                        "Incapable of retrieving header from non-existent parent of "
+                            + block.getHeader().getNumber()
+                            + "."));
+
+    final ProtocolSpec<C> protocolSpec =
+        protocolSchedule.getByBlockNumber(block.getHeader().getNumber());
+    final BlockHeaderValidator<C> blockHeaderValidator = protocolSpec.getBlockHeaderValidator();
 
     // Import block
     final PersistBlockTask<C> importTask =
         PersistBlockTask.create(
-            protocolSchedule, protocolContext, block, HeaderValidationMode.FULL, metricsSystem);
+            protocolSchedule, protocolContext, block, HeaderValidationMode.NONE, metricsSystem);
```

Comment:
Can we create this inside the if(validate) block so it's only created if it will actually be used? Mostly just so that the first thing you come across isn't `HeaderValidationMode.NODE` which is a bit scary. :)